conans/client/generators/compiler_args.py

```diff
@@ -64,6 +65,8 @@ def content(self):
         flags.extend(self._deps_build_info.sharedlinkflags)
         flags.extend(self._deps_build_info.exelinkflags)
         flags.extend(self._libcxx_flags())
+        flags.extend(format_frameworks(self._deps_build_info.frameworks, compiler=self.compiler))
```

Comment:
This will break gcc on linux, right? Probably it has to be applied only when `the_os` os is `Macos`? I know the recipe might declare only the cppinfo entry when Macos, but should we double-check?

Reply:
yeah, I think the sanity check could be added. however, will it work for header-only recipe? I need to check that first